From 99e5f4316c5ac1045d6a4c97e51a18a69a1ead06 Mon Sep 17 00:00:00 2001
From: Gabriel Craciunescu <crazy@frugalware.org>
Date: Tue, 4 Oct 2016 11:05:57 +0100
Subject: [PATCH 05/19] Fix usage of qdbus variable in startkde

$qdbus is a full path to the qdbus binary, not relying on it being in
$PATH

It was completely unset in startplasma and startkde made a call to
ksplash using qdbus assuming it's in the path

Reviewed-by: David Edmundson
---
 startkde/startkde.cmake    |  2 +-
 startkde/startplasma.cmake | 14 +++++++++++++-
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/startkde/startkde.cmake b/startkde/startkde.cmake
index b320234..ddd3663 100644
--- a/startkde/startkde.cmake
+++ b/startkde/startkde.cmake
@@ -359,7 +359,7 @@ if test $? -ne 0; then
   exit 1
 fi
 
-qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit
+$qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit
 
 # finally, give the session control to the session manager
 # see kdebase/ksmserver for the description of the rest of the startup sequence
diff --git a/startkde/startplasma.cmake b/startkde/startplasma.cmake
index fa20ef4..2904e8f 100644
--- a/startkde/startplasma.cmake
+++ b/startkde/startplasma.cmake
@@ -49,6 +49,18 @@ if test -z "$dl"; then
   esac
 fi
 
+# in case we have been started with full pathname spec without being in PATH
+bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
+if [ -n "$bindir" ]; then
+  qbindir=`qtpaths --binaries-dir`
+  qdbus=$qbindir/qdbus
+  case $PATH in
+    $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
+    *) PATH=$bindir:$PATH; export PATH;;
+  esac
+else
+  qdbus=qdbus
+fi
 
 # Activate the kde font directories.
 #
@@ -128,7 +140,7 @@ if test $? -ne 0; then
   exit 1
 fi
 
-qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit
+$qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit
 
 # finally, give the session control to the session manager
 # see kdebase/ksmserver for the description of the rest of the startup sequence
-- 
2.7.4

